app/components/SearchDropdown.tsx

```diff
@@ -13,16 +12,30 @@ interface Props {
   errorSchema?: any;
 }
 
-const SearchDropdown: React.FunctionComponent<Props> = ({
+export const SearchDropdownComponent: React.FunctionComponent<Props> = ({
   id,
   placeholder,
   defaultInputValue,
   options,
   selected,
   inputProps,
-  onChange,
-  errorSchema
+  onChange
 }) => {
+  // Get all indexes that contain undefined
+  const getAllUndefinedIndexes = (array, value) => {
+    const indexArray = [];
+    array.forEach((item, index) => {
+      if (item === value) indexArray.push(index);
+    });
+    return indexArray;
+  };
+
+  // Remove all undefined indexes from options (required so archived/deprecated products do not show as options in the dropdown)
+  const undefinedIndexes = getAllUndefinedIndexes(options, undefined);
+  for (let i = undefinedIndexes.length - 1; i >= 0; i--) {
+    options.splice(undefinedIndexes[i], 1);
+  }
+
```

Comment:
Props should be immutable, `options.splice` breaks that.
This could whole thing could also be written as `options.filter(o => o !== undefined)`.
Either way I don't think that handling some product specific things in here is the right solution.